src/Datadog.Trace.ClrProfiler.Managed/Emit/ObjectExtensions.cs

```diff
@@ -68,16 +68,19 @@ public static MemberResult<object> CallMethod<TArg1>(this object source, string
         /// <returns><c>true</c> if the property exists, otherwise <c>false</c>.</returns>
         public static bool TryGetPropertyValue<TResult>(this object source, string propertyName, out TResult value)
         {
-            var type = source.GetType();
+            if (source != null)
```

Comment:
I would prefer to throw `ArgumentNullException` when `source` is null. Extensions methods are used as if they were instance methods and should act like them. This is just weird:

```csharp
object foo = null;
foo.TryGetPropertyValue(...); // should throw ArgumentNullException
```

This is exactly why I added a wrapper `MemberResult<T>` class to the newer implementation and return `MemberResult<T>.NotFound` instead of null.

Reply:
The thought was to have a higher level guard against crashing client applications. 
It seemed like a reasonable place to guard against null reference exceptions, especially when the intent of the method is to "try" to get a property value.
It feels like an easy guard to increase stability.

Reply:
Ok, that makes sense. Let's leave it.